pkg/restore/util.go

```diff
@@ -524,3 +524,23 @@ func ZapRanges(ranges []rtree.Range) []zapcore.Field {
 		zap.Uint64("total size", totalSize),
 	}
 }
+
+func parseQuoteName(name string) (string, string) {
+	schemaRune := make([]rune, 0, len(name))
+	for i, c := range name {
+		schemaRune = append(schemaRune, c)
+		if i == 0 {
+			continue
+		}
+		if name[i] == '`' && name[i-1] != '`' {
+			break
+		}
+	}
+	schema := string(schemaRune)
+	return unQuoteName(schema), unQuoteName(strings.TrimPrefix(name, schema+"."))
+}
+
+func unQuoteName(name string) string {
+	name = strings.TrimLeft(name, "`")
+	return strings.TrimRight(name, "`")
+}
```

Comment:
i don't think this implementation produces the correct unquoting when the table name contains interior `` ` ``.

```go
schema, table := parseQuoteName("`a``b`.``````")
c.Assert(schema, Equals, "a`b")
c.Assert(table, Equals, "``")
```

Reply:
yes, the corner case seems not easy. do you have some advices? there already have the solution.

Reply:
You could regexp match for ``` `(?:[^`]|``)*` ``` and extract each part of name with ```strings.ReplaceAll(result[1:len(result)-1], "``", "`")```.

Reply:
Done, PTAL again